src/modules/ModuleRunner.ts

```diff
@@ -17,28 +17,118 @@ limitations under the License.
 import { safeSet } from "matrix-js-sdk/src/utils";
 import { TranslationStringsObject } from "@matrix-org/react-sdk-module-api/lib/types/translations";
 import { AnyLifecycle } from "@matrix-org/react-sdk-module-api/lib/lifecycles/types";
-import { AllExtensions } from "@matrix-org/react-sdk-module-api/lib/types/extensions";
-import { DefaultCryptoSetupExtensions } from "@matrix-org/react-sdk-module-api/lib/lifecycles/CryptoSetupExtensions";
-import { DefaultExperimentalExtensions } from "@matrix-org/react-sdk-module-api/lib/lifecycles/ExperimentalExtensions";
-import { RuntimeModule } from "@matrix-org/react-sdk-module-api";
+import {
+    DefaultCryptoSetupExtensions,
+    ProvideCryptoSetupExtensions,
+} from "@matrix-org/react-sdk-module-api/lib/lifecycles/CryptoSetupExtensions";
+import {
+    DefaultExperimentalExtensions,
+    ProvideExperimentalExtensions,
+} from "@matrix-org/react-sdk-module-api/lib/lifecycles/ExperimentalExtensions";
 
 import { AppModule } from "./AppModule";
 import { ModuleFactory } from "./ModuleFactory";
 
 import "./ModuleComponents";
 
+class ExtensionImplementationMap {
+    public hasDefaultCryptoSetupExtension: boolean = true;
+    public hasDefaultExperimentalExtension: boolean = true;
+}
+
+/**
+ * Handles and manages any extensions provided by modules
+ */
+class ExtensionsManager {
+    private _cryptoSetup: ProvideCryptoSetupExtensions;
+    private _experimental: ProvideExperimentalExtensions;
+    private _implementionMap: ExtensionImplementationMap = new ExtensionImplementationMap();
+
+    /**
+     * Create a new instance
+     */
+    public constructor() {
+        // Set up defaults
+        this._cryptoSetup = new DefaultCryptoSetupExtensions();
+        this._experimental = new DefaultExperimentalExtensions();
+    }
+
+    /**
+     * Provides a crypto setup extension.
+     *
+     * @returns The registered extension. If no module provides this extension, a default implementation is returned
+     */
+    public get cryptoSetup(): ProvideCryptoSetupExtensions {
+        return this._cryptoSetup;
+    }
+
+    /**
+     * Provides a n experimental extension.
+     *
+     * @remarks
+     * This method extension is provided to simplify experimentaion an development, and is not intended for production code
+     *
+     * @returns The registered extension. If no module provides this extension, a default implementation is returned
+     */
+    public get experimental(): ProvideExperimentalExtensions {
+        return this._experimental;
+    }
+
+    /**
+     * Resets the extension to the defaults
```

Comment:
```suggestion
     * Reset the extensions to the defaults.
```

[It would also be nice to be consistent about whether we are using the imperative "Reset the extensions..." or the indicative "Resets the extensions..."].